### ISSUE TITLE
How do you suggest implementing multiple users?

### ISSUE THREAD

**Issue by cgillenscs** [dgreif/ring-alarm#22]
First I want to thank you for this plugin. It works great with the Apple Watch since the watch verifies identity without verification on the phone.

Here's my application: I've set up and want to continue to manage my home through the iOS/macOS Home app. (I've got a few other plugins too.)  I'd like to just "Invite" my wife  in the Home app so I can continue to manage the whole thing and she can just use it and be oblivious to my tinkering.

When she uses this plugin to, say, disarm the alarm, she's prompted that there needs to be the correct user (me) logged into her phone's app.

Is there any way to use this plugin with multiple Ring users?

The two workarounds I've thought of are to:
1. Obviously, I could just let her use my Ring account on her phone. Or,
2. I could run a second homebridge instance just for her.

Both are kind of clunky, and what if I want to add another person, like a grandparent or nanny?

Anyway have you put any thought into this? I've looked through the docs and issues and can't see any mention of this type of thing.

Thanks again.

**Comment by dgreif** [comment]
@cgillenscs I totally understand your situation and I have the same setup between myself and my wife.  There shouldn't be any need to deal with multiple Ring users.  In fact, she shouldn't even be required to have the Ring app on her phone for the HomeKit plugin to work.  What you need to do is set up family sharing at the iOS level.  On your iPhone, go to Settings -> Click on your name at the very top (says Apple ID, iCloud, etc. under it) -> Family Sharing -> Add Family Member.  Once you go through the process, all of your devices from HomeKit will be shared between your two accounts.  She wouldn't need to add the homebridge on her phone, she would just be connected to it through your account.  There are tons of other benefits of setting up family sharing, like sharing app purchases and location sharing (unless you don't want to be able to track each other 😄).  The other awesome benefit within HomeKit is that you can set up automation based on both of you being home/away.  The ideal setup is that when _both_ of you leave the house, the alarm sets to away.  When _either_ of you get home, it disarms.  This might get complicated if you have others of driving in your home, but for a two-adult-household it works perfectly.  See https://github.com/dgreif/ring-alarm/tree/master/homebridge#changes-modes-on-arriveleave-home for detailed instructions on how to get around HomeKit's requirement to click a button to arm/disarm the alarm.

Hope that's helpful enough to get you on your way.  If you need any clarification, or if I overlooked some issue in your set up, I'm happy to answer more questions.

**Comment by cgillenscs** [comment]
OK I got it working in the way you describe, but not via those settings. I've always had Family Sharing active and her in my family, but the home/rooms/devices only show up on my account. But then I (as I had done before) added my wife via "Inviting" her to the home via the Home app's setting. I swear when I tried this about 6 months ago, she would get a dialog stating she needed to be logged in as me in her Ring app.  But hey, it works exactly as you said now! Maybe something changed or I am just mis-remembering.

Thanks again!
